File: app/src/main/java/it/droidcon/testingdaggerrxjava/dagger/ApplicationComponent.java
package it.droidcon.testingdaggerrxjava.dagger;

import javax.inject.Singleton;

import dagger.Component;
import it.droidcon.testingdaggerrxjava.userlist.UserListComponent;
import it.droidcon.testingdaggerrxjava.userlist.UserListModule;

@Singleton
@Component(modules = {
        UserInteractorModule.class,
        StackOverflowServiceModule.class
})
public interface ApplicationComponent {
    UserListComponent userListComponent(UserListModule module);
}

File: app/src/main/java/it/droidcon/testingdaggerrxjava/dagger/ApplicationComponent.kt
package it.droidcon.testingdaggerrxjava.dagger

import dagger.Component
import it.droidcon.testingdaggerrxjava.userlist.UserListComponent
import it.droidcon.testingdaggerrxjava.userlist.UserListModule
import javax.inject.Singleton

@Singleton
@Component(modules = arrayOf(StackOverflowServiceModule::class))
interface ApplicationComponent {
    fun userListComponent(module: UserListModule): UserListComponent
}
